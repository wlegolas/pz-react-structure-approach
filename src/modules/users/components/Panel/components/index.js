export { default as Search } from './Search/Search'
export { default as UserInfo } from './UserInfo/UserInfo'
