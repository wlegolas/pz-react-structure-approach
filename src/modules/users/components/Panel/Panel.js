import PropTypes from 'prop-types'
import React from 'react'
import { Search, UserInfo } from './components'

const Panel = ({ userInfo }) => (
  <div className="pz-user-panel">
    <Search onKeyUp={() => {}} />
    <UserInfo userInfo={userInfo} />
  </div>
)

Panel.propTypes = {
  userInfo: PropTypes.object.isRequired
}

export default Panel
