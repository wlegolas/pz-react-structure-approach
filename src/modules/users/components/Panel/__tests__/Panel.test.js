import toJson from 'enzyme-to-json'
import React from 'react'
import Panel from '../Panel'

describe('<Panel />', () => {
  let wrapper

  beforeAll(() => {
    wrapper = shallow(<Panel />)
  })

  describe('render()', () => {
    it('should render the component', () => {
      expect(toJson(wrapper)).to.matchSnapshot()
    })
  })

  describe('verify the structure', () => {
    it('should have a section with filter component', () => {
      expect(wrapper.find('.pz-search')).to.have.lengthOf(1)
    })

    it('should have a section with user info', () => {
      expect(wrapper.find('.pz-user-container')).to.have.lengthOf(1)
    })
  })
})
