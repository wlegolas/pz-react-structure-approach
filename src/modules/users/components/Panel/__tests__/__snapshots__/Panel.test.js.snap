// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> render() should render the component 1`] = `
<Panel
  userInfo={
    Object {
      "avatarUrl": "https://avatars2.githubusercontent.com/u/698363?v=4",
      "name": "Weslley Alves de Oliveira",
      "totalFollowers": 5,
      "totalFollowing": 10,
    }
  }
>
  <div
    className="pz-user-panel"
  >
    <Search
      onKeyUp={[Function]}
    >
      <div
        className="pz-search"
      >
        <input
          onKeyUp={[Function]}
          placeholder="Fill the github user and press enter to search"
          type="text"
        />
      </div>
    </Search>
    <UserInfo
      userInfo={
        Object {
          "avatarUrl": "https://avatars2.githubusercontent.com/u/698363?v=4",
          "name": "Weslley Alves de Oliveira",
          "totalFollowers": 5,
          "totalFollowing": 10,
        }
      }
    >
      <div
        className="pz-user-container"
      >
        <div
          className="pz-user-info"
        >
          <div
            className="pz-box pz-left-box"
          >
            <img
              alt="GiHhub avatar"
              src="https://avatars2.githubusercontent.com/u/698363?v=4"
            />
          </div>
          <div
            className="pz-box pz-right-box"
          >
            <h1>
              Weslley Alves de Oliveira
            </h1>
            <p
              className="pz-label"
            >
              Following: 
              10
            </p>
            <p
              className="pz-label"
            >
              Followers: 
              5
            </p>
          </div>
        </div>
      </div>
    </UserInfo>
  </div>
</Panel>
`;
